pkg/controller/policyrecommendation/policyrecommendation_controller.go

```diff
@@ -0,0 +1,452 @@
+	tierWatchReady := &utils.ReadyFlag{}
+
+	// Create the reconciler
+	reconciler := newReconciler(mgr, opts, licenseAPIReady, tierWatchReady)
+
+	// Create a new controller
+	controller, err := controller.New(render.PolicyRecommendationControllerName, mgr,
+		controller.Options{
+			Reconciler: reconcile.Reconciler(reconciler),
+		})
+	if err != nil {
+		return err
+	}
+
+	k8sClient, err := kubernetes.NewForConfig(mgr.GetConfig())
+	if err != nil {
+		log.Error(err, "Failed to establish a connection to k8s")
+		return err
+	}
+
+	go utils.WaitToAddLicenseKeyWatch(controller, k8sClient, log, licenseAPIReady)
+	go utils.WaitToAddTierWatch(networkpolicy.TigeraComponentTierName, controller, k8sClient, log, tierWatchReady)
+	go utils.WaitToAddNetworkPolicyWatches(controller, k8sClient, log, []types.NamespacedName{
+		{Name: render.PolicyRecommendationPolicyName, Namespace: render.PolicyRecommendationNamespace},
+		{Name: networkpolicy.TigeraComponentDefaultDenyPolicyName, Namespace: render.PolicyRecommendationNamespace},
+	})
+
+	return add(mgr, controller)
+}
+
+// newReconciler returns a new *reconcile.Reconciler.
+func newReconciler(
+	mgr manager.Manager,
+	opts options.AddOptions,
+	licenseAPIReady *utils.ReadyFlag,
+	tierWatchReady *utils.ReadyFlag,
+) reconcile.Reconciler {
+
+	r := &ReconcilePolicyRecommendation{
+		client:          mgr.GetClient(),
+		scheme:          mgr.GetScheme(),
+		provider:        opts.DetectedProvider,
+		status:          status.New(mgr.GetClient(), "policy-recommendation", opts.KubernetesVersion),
+		clusterDomain:   opts.ClusterDomain,
+		licenseAPIReady: licenseAPIReady,
+		tierWatchReady:  tierWatchReady,
+		usePSP:          opts.UsePSP,
+	}
+
+	r.status.Run(opts.ShutdownContext)
+
+	return r
+}
+
+// add adds watches for resources that are available at startup.
+func add(mgr manager.Manager, c controller.Controller) error {
+	var err error
+
+	// Watch for changes to primary resource PolicyRecommendation
+	err = c.Watch(&source.Kind{Type: &operatorv1.PolicyRecommendation{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	if err = utils.AddNetworkWatch(c); err != nil {
+		return fmt.Errorf("policy-recommendation-controller failed to watch Network resource: %w", err)
+	}
+
+	if err = imageset.AddImageSetWatch(c); err != nil {
+		return fmt.Errorf("policy-recommendation-controller failed to watch ImageSet: %w", err)
+	}
+
+	if err = utils.AddAPIServerWatch(c); err != nil {
+		return fmt.Errorf("policy-recommendation-controller failed to watch APIServer resource: %w", err)
+	}
+
+	// Watch the given secrets in each both the policy-recommendation and operator namespaces
+	for _, namespace := range []string{common.OperatorNamespace(), render.PolicyRecommendationNamespace} {
+		for _, secretName := range []string{
+			render.TigeraElasticsearchGatewaySecret, render.ElasticsearchPolicyRecommendationUserSecret,
+			render.ManagerInternalTLSSecretName, certificatemanagement.CASecretName,
```

Comment:
I'm not 100% sure if we need `render.ManagerInternalTLSSecretName`, since there is not direct connection to manager and vice versa to/from PolicyRecController - all the communication is through Calico Resources

but maybe @rene-dekker will know better?

Reply:
Sounds right if the controller does not make any calls to voltron.

Although the contents are the same, use relasticsearch.PublicCertSecret rather than render.TigeraElasticsearchGatewaySecret, which is meant for trusting es-gateway.